### PR TITLE
[#16][#17] [UI][Integrate] As a user, I can see the loading screen for surveys

### PR DESCRIPTION
**Pull request by Tuubz**
Closes https://github.com/nimblehq/ic-flutter-taher-toby/issues/16
Closes https://github.com/nimblehq/ic-flutter-taher-toby/issues/17

## What happened 👀

- Add a skeleton widget to indicate the loading process of the surveys on the home screen
- Add [shimmer](https://pub.dev/packages/shimmer), for the loading animations

## Proof Of Work 📹

Please note that the emulator is a bit laggy

**SUCCESS CASE:** 

https://user-images.githubusercontent.com/18277915/229420152-bbd37cef-7d07-4aed-87da-10f0434dadc7.mov

**ERROR CASE:** 

https://user-images.githubusercontent.com/18277915/229420352-cbe9f01d-acf6-4cb6-a256-283d74ce2c5d.mov

### CODE REVIEW

**Comment by luongvo** [comment]
@Tuubz please add some line breaks to show the preview of your POWs

image

**Review by luongvo — APPROVED** [review]

**Comment by Tuubz** [comment]
@Shayokh144 Updated and resolved conflicts 🚀

File: lib/ui/home/widget/home_skeleton_loading.dart [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+              _buildSkeleton(width: screenWidth / 3),
+            ],
+          ),
+        ),
+        Padding(
+          padding: const EdgeInsets.only(top: AppDimensions.spacing18),
+          child: _buildSkeleton(
+              width: AppDimensions.homeAvatarSize,
+              height: AppDimensions.homeAvatarSize,
+              borderRadius: AppDimensions.homeAvatarSize),
```
Comment (Shayokh144) [inline lib/ui/home/widget/home_skeleton_loading.dart:61]:
```suggestion
              borderRadius: AppDimensions.homeAvatarSize,),
```

File: test/ui/home/home_view_model_test.dart [hunk 2]
```diff
@@ -55,10 +55,15 @@ void main() {
...
-      homeViewModel.loadSurveys();
-
       expect(surveysStream, emitsInOrder([surveys]));
-      expect(stateStream, emitsInOrder([const HomeState.loadSurveysSuccess()]));
+      expect(
+          stateStream,
+          emitsInOrder([
+            const HomeState.loading(),
+            const HomeState.loadSurveysSuccess(),
+          ]));
```
Comment (Shayokh144) [inline test/ui/home/home_view_model_test.dart:64]:
```suggestion
          ]),);
```

**Pull request merged**